modules/xfem/src/base/XFEMGeometricCut3D.C

```diff
@@ -38,65 +39,51 @@ XFEMGeometricCut3D::cutElementByGeometry(const Elem* elem,
 {
   bool cut_elem = false;
 
-   const int hex_indices[6][4] = {{0,3,2,1},{0,1,5,4},{1,2,6,5},{2,3,7,6},{3,0,4,7},{4,5,6,7}};
-   const int tet_indices[4][3] = {{0,2,1},{0,1,3},{1,2,3},{2,0,3}};
+  for (unsigned int i = 0; i < elem->n_sides(); ++i)
+  {
+    // This returns the lowest-order type of side.
+    UniquePtr<Elem> curr_side = elem->side(i);
+    if (curr_side->dim() != 2)
+      mooseError("In cutElementByGeometry dimension of side must be 2, but it is " << curr_side->dim());
+    unsigned int n_edges = curr_side->n_sides();
 
-   unsigned int n_face_nodes = 0;
-   if (elem->n_nodes() == 8)
-     n_face_nodes = 4;
-   else if (elem->n_nodes() == 4)
-     n_face_nodes = 3;
-   else
-     mooseError("this method only works for linear hexes and tets");
+    std::vector<unsigned int> cut_edges;
+    std::vector<Real> cut_pos;
 
-    for (unsigned int i = 0; i < elem->n_sides(); ++i)
+    for (unsigned int j = 0; j < n_edges; j++)
     {
-      std::vector<unsigned int> cut_edges;
-      std::vector<Real> cut_pos;
-
-      for (unsigned int j=0; j < n_face_nodes; j++)
+      // This returns the lowest-order type of side.
+      UniquePtr<Elem> curr_edge = curr_side->side(j);
+      if (curr_edge->type() != EDGE2)
+        mooseError("In cutElementByGeometry face edge must be EDGE2, but type is: " << libMesh::Utility::enum_to_string(curr_edge->type())
+                   << " base element type is: " << libMesh::Utility::enum_to_string(elem->type()));
+      Node * node1 = curr_edge->get_node(0);
+      Node * node2 = curr_edge->get_node(1);
+      Point p1((*node1)(0), (*node1)(1), (*node1)(2));
+      Point p2((*node2)(0), (*node2)(1), (*node2)(2));
```

Comment:
Node is inherited from Point. You don't need to construct a Point if you have a Node.